File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& v) {
        set<int> st(v.begin(), v.end()); // Store all elements in a set
        int maxseq = 0;

        for (int num : v) {
            if (st.find(num - 1) == st.end()) { // Only start from the smallest in a sequence
                int cur_num = num;
                int seq = 1;

                while (st.find(cur_num + 1) != st.end()) {
                    cur_num++;
                    seq++;
                }
                maxseq = max(maxseq, seq);
            }
        }
        return maxseq;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            set.add(num);
        }
        int longestConsecutiveSequence = 0;
        for (int num : nums) {
            if (!set.contains(num - 1)) {
                int currentNumber = num;
                int currentConsecutiveSequence = 1;
                while (set.contains(currentNumber + 1)) {
                    currentNumber++;
                    currentConsecutiveSequence++;
                }
                longestConsecutiveSequence = Math.max(longestConsecutiveSequence, currentConsecutiveSequence);
            }
        }
        return longestConsecutiveSequence;
    }
}